src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -322,19 +322,30 @@ void SingleStreamDecoder::scanFileAndUpdateMetadataAndIndex() {
 void SingleStreamDecoder::readCustomFrameMappingsUpdateMetadataAndIndex(
     int streamIndex,
     FrameMappings customFrameMappings) {
-  auto& all_frames = customFrameMappings.all_frames;
-  auto& is_key_frame = customFrameMappings.is_key_frame;
-  auto& duration = customFrameMappings.duration;
+  const torch::Tensor& all_frames =
+      customFrameMappings.all_frames.to(torch::kLong);
+  const torch::Tensor& is_key_frame =
+      customFrameMappings.is_key_frame.to(torch::kBool);
+  const torch::Tensor& duration = customFrameMappings.duration.to(torch::kLong);
   TORCH_CHECK(
       all_frames.size(0) == is_key_frame.size(0) &&
           is_key_frame.size(0) == duration.size(0),
       "all_frames, is_key_frame, and duration from custom_frame_mappings were not same size.");
 
+  // Allocate vectors using num frames to reduce reallocations
+  int64_t numFrames = all_frames.size(0);
+  streamInfos_[streamIndex].allFrames.reserve(numFrames);
+  streamInfos_[streamIndex].keyFrames.reserve(numFrames);
+  // Access tensor data directly rather than element wise to speed up loop below
+  auto* pts_data = all_frames.data_ptr<int64_t>();
+  auto* is_key_frame_data = is_key_frame.data_ptr<bool>();
+  auto* duration_data = duration.data_ptr<int64_t>();
```

Comment:
Ah, great catch. Instead of using raw pointers, let's use accessors instead, they're safer https://docs.pytorch.org/cppdocs/notes/tensor_basics.html#cpu-accessors